### DISCUSSION TITLE
Filter top level resource by embedded resource while keeping embedded resource unfiltered

### DISCUSSION THREAD

**Discussion by Evaldas-B** [supabase/supabase#8385] [Questions] [answered]
As described in [PostgREST docs](https://postgrest.org/en/stable/api.html#embedding-with-top-level-filtering) I  am trying to apply a filter on a top-level resource by filtering embedded resource. Just like in the docs example schema consists of **films** and **actors** that have many-to-many relationship (junction table **rows**)

![image](https://user-images.githubusercontent.com/53705834/184894165-45d90e09-71f5-411f-bd59-eb970c2939e0.png)

```SQL
CREATE TABLE films (
	id INT PRIMARY KEY,
  	title TEXT
);

CREATE TABLE actors (
	id INT PRIMARY KEY,
  	name TEXT
);

CREATE TABLE roles (
	film_id INT REFERENCES films(id),
	actor_id INT REFERENCES actors(id)
);


INSERT INTO films (id, title) VALUES (1, "Mad Max: Fury Road"), (2, "Legend"), (3, "Monster");
INSERT INTO actors (id, name) VALUES (1, "Tom Hardy"), (2, "Charlize Theron");
INSERT INTO roles (film_id, actor_id) VALUES (1, 1), (1, 2), (2, 1), (3, 2);
```
When filtering by the embedded resource all films are returned but actors are filtered:
```Javascript
const { data: films, error } = await supabaseClient
    .from("films")
    .select(`*, actors(name)`)
    .like("actors.name", "Tom Hardy")
```

<details>
  <summary>Query result</summary>

  ```json
  [
    {
      "id": 1,
      "title": "Mad Max: Fury Road",
      "actors": [
        {
          "name": "Tom Hardy"
        }
      ]
    },
    {
      "id": 2,
      "title": "Legend",
      "actors": [
        {
          "name": "Tom Hardy"
        }
      ]
    },
    {
      "id": 3,
      "title": "Monster",
      "actors": []
    }
  ]
  ```
</details>

When filtering by the top-level resource **(By using !inner)** only the films that have a specified actor are returned
```js
const { data: films, error } = await supabaseClient
    .from("films")
    .select(`*, actors!inner(name)`)
    .like("actors.name", "Tom Hardy")
```
<details>
  <summary>Query result</summary>

```json
[
  {
    "id": 1,
    "title": "Mad Max: Fury Road",
    "actors": [
      {
        "name": "Tom Hardy" 
         // Expected to have "Charlize Theron" in this list too
      }
    ]
  },
  {
    "id": 2,
    "title": "Legend",
    "actors": [
      {
        "name": "Tom Hardy"
      }
    ]
  }
]
```
</details>

The problem is that the filter is applied both on the top-level resource and embedded resource. Is it possible to write a Supabase query that only filters the top-level resource without filtering the nested resource? In other words, list all the films that have a specified actor while at the same time listing all the actors that starred in the returned films

**Comment by GaryAustin1** [comment, marked as answer]
I think you can do this with an alias on the embedded resource in addition to the embedded resource then filter on the alias...
https://github.com/supabase/supabase/discussions/6587#discussioncomment-2657550

**Reply by Evaldas-B** [reply]
Yes, that works.

in this specific case the query looks like this:
```js
const { data: films, error } = await supabaseClient
    .from("films")
    .select(`*, actors(name), temp_actors:actors!inner(name)`)
    .like("temp_actors.name", "Tom Hardy")
```

<details>
  <summary>Query result</summary>

```json
[
  {
    "id": 1,
    "title": "Mad Max: Fury Road",
    "temp_actors": [
      {
        "name": "Tom Hardy"
      }
    ],
    "actors": [
      {
        "name": "Tom Hardy"
      },
      {
        "name": "Charlize Theron"
      }
    ]
  },
  {
    "id": 2,
    "title": "Legend",
    "temp_actors": [
      {
        "name": "Tom Hardy"
      }
    ],
    "actors": [
      {
        "name": "Tom Hardy"
      }
    ]
  }
]
```
</details>
